### PR TITLE
No need to pass GOFLAGS to $goflags variables with go 1.11

### LINKED ISSUE #70062: Build breaks when using GOFLAGS="-p=1"
> /kind bug

**What happened**:
When building kubernetes from source for 1.10 it was possible to limit the parallel linker calls using GOFLAGS="-p 1". (I understand that the new format is GOFLAGS="-p=1" so I am using that)
With kubernetes 1.11 the build aborts:

```
go test: p flag may be set only once
run "go help test" or "go help testflag" for more information
!!! [1021 19:40:30] Call tree:
!!! [1021 19:40:30]  1: /home/tkruse/kubernetes/src/kubernetes-1.12.0/hack/lib/golang.sh:735 kube::golang::build_binaries_for_platform(...)
!!! [1021 19:40:30]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [1021 19:40:30] Call tree:
!!! [1021 19:40:30]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
!!! [1021 19:40:30] Call tree:
!!! [1021 19:40:30]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
make: *** [Makefile:92: all] Error 1
==> ERROR: A failure occurred in build().
    Aborting...
```

**What you expected to happen**:
The build should complete, using just one process for linking.

**How to reproduce it (as minimally and precisely as possible)**:
export GOFLAGS="-p=1"
make -j1

**Anything else we need to know?**:
The environment is quite memory constrained, therefore preventing the build to fork a linker for each CPU is required. Any other option instead of the GOFLAGS would work as well.

**Environment**:
- Kubernetes version (use `kubectl version`):\
1.12.0
- OS (e.g. from /etc/os-release):
Arch Linux ARM

- Others:
go version go1.11.1 linux/arm64

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

Do not save `GOFLAGS` in `$goflags`. 

In go 1.11, go commands will use `GOFLAGS` as default flags, see
https://golang.org/doc/go1.11#go_command.

There is no need to pass GOFLAGS to $goflags, and if we do, go commands
will fail with "duplicate flags" error, e.g.

```
$ make test-integration WHAT=./test/integration/scheduler GOFLAGS="-v"
...
go test: v flag may be set only once
run "go help test" or "go help testflag" for more information
...
```

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #70062 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cofyc** [comment]
ping @cblecker in case you missed it

**Comment by cblecker** [comment]
This is still on my radar, but I'm caught up with a bunch of KubeCon prep so my response will be slightly delayed. I want to run some testing to ensure this doesn't break other parts of the build system.

**Comment by ncdc** [comment]
I just ran into this and would love to see it get merged - @cblecker do you think you'll be able to get around to this soon? Thanks!

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by cblecker** [comment]
/test pull-kubernetes-godeps
/approve

This works for me locally. We'll have to keep a close eye if there are any side effects, but reviewing test outputs it looks like it takes.

cc: @ixdy

File: hack/lib/golang.sh [hunk 1]
```diff
@@ -677,9 +677,7 @@ kube::golang::build_binaries() {
     local host_platform
     host_platform=$(kube::golang::host_platform)
 
-    # Use eval to preserve embedded quoted strings.
     local goflags goldflags goasmflags gogcflags
```
Comment (ixdy) [inline hack/lib/golang.sh]: we can also probably remove the `goflags` local variable and then remove all references to `$goflags`?

Comment (ixdy) [reply hack/lib/golang.sh]: oh, scratch that

**Pull request merged**